public/app/features/scopes/selector/ScopesSelectorService.ts

```diff
@@ -67,16 +67,23 @@ export class ScopesSelectorService extends ScopesServiceBase<ScopesSelectorServi
     // Making a copy as we will be changing this in place and then updating state later.
     // This though does not make a deep copy so you cannot rely on reference of nested nodes changing.
     const nodes = { ...this.state.nodes };
-    let currentLevel: NodesMap = nodes;
+    let parentNode: Node | undefined = nodes[''];
     let loadingNodeName = path[0];
+    let currentNode = nodes[''];
 
     if (path.length > 1) {
-      const pathToParent = path.slice(0, path.length - 1);
-      currentLevel = getNodesAtPath(nodes, pathToParent);
+      const pathToParent = path.slice(1, path.length - 1);
+      parentNode = getNodesAtPath(nodes[''], pathToParent);
```

Comment:
Changed a bit how `getNodesAtPath` work so I could reuse it and so needed to change this a bit